functions/apiProxy/participant_data_handler.py

```
@@ -0,0 +1,875 @@
+            'can_access_pii': security_level.value in ['premium', 'enterprise'],  # Enhanced PII access for higher security levels
+            'can_access_grades': security_level.value == 'enterprise',  # Only enterprise can access grades
+            'course_access': []  # Empty means access to all funded courses
+        }
+        
+    except (IndexError, ValueError) as e:
+        if "Database" in str(e):
+            raise e
+        raise ValueError("Invalid API key format")
+    except Exception as e:
+        logging.error(f"API key authentication error: {str(e)}")
+        raise ValueError("Authentication failed")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)